keps/sig-network/1860-kube-proxy-IP-node-binding/README.md

```diff
@@ -149,3 +159,284 @@ On downgrade, the feature gate will simply be disabled, and as long as `kube-pro
 ### Version Skew Strategy
 
 Version skew from the control plane to `kube-proxy` should be trivial since `kube-proxy` will simply ignore the `ipMode` field.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: LoadBalancerIPMode
+  - Components depending on the feature gate: kube-proxy, kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It works. The forwarding rules for services which have the value of `ipMode` had been set to "Proxy" will be removed by kube-proxy.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes. There are some unit tests and an integration test added for this feature enablement/disablement.
```

Comment:
Are there? Or will be added?

Reply:
These tests have added in the implemention(https://github.com/kubernetes/kubernetes/pull/118895).

Reply:
Please correct me if I'm wrong, but I don't see enablement/disablement features there.
I see tests that exercise the feature-gate being on and off, but I don't see tests that change it in the middle of the test [which is what the enablement/disablement is doing.]


On unrelated note - how the implementation PR got merged, given the feature didn't went through PRR and wasn't tracked as part of the release:
https://github.com/orgs/kubernetes/projects/140/views/1?filterQuery=

@thockin ^^

Reply:
> On unrelated note - how the implementation PR got merged, given the feature didn't went through PRR and wasn't tracked as part of the release

I think the KEP was originally approved to be implemented before PRR existed (and then just didn't get implemented). I'm not sure what that means process-wise.

Reply:
@wojtek-t  The test `TestUpdateServiceLoadBalancerStatus` in pkg/registry/core/service/storage/storage_test.go did this.
https://github.com/RyanAoh/kubernetes/blob/e6863757f4fd9b6997ca38dea549e3b370463db9/pkg/registry/core/service/storage/storage_test.go#L11935-L11947

Reply:
> On unrelated note - how the implementation PR got merged, given the feature didn't went through PRR and wasn't tracked as part of the release: https://github.com/orgs/kubernetes/projects/140/views/1?filterQuery=

@wojtek-t https://github.com/kubernetes/kubernetes/pull/118895 was reverted by https://github.com/kubernetes/kubernetes/pull/119876 for release resons. And I have opened a new one https://github.com/kubernetes/kubernetes/pull/119937, we can make this run into the right process.